pkg/cloudprovider/servicelb.go

```diff
@@ -597,6 +600,16 @@ func (k *k3s) newDaemonSet(svc *core.Service) (*apps.DaemonSet, error) {
 		ds.Labels[nodeSelectorLabel] = "true"
 	}
 
+	// Fetch tolerations from the "svccontroller.k3s.cattle.io/tolerations" annotation on the service
+	// and append them to the DaemonSet's pod tolerations.
+	tolerations, err := k.getTolerations(svc)
+	if err != nil {
+		return nil, err
+	}
+	if len(tolerations) > 0 {
```

Comment:
nit: you can just append them, if the length is 0 then the append is a no-op anyway.

Reply:
Updated!